scipy/sparse/tests/test_base.py

```diff
@@ -1294,6 +1294,20 @@ def check_equal_but_not_same_array_attribute(attribute):
                 for attribute in ('offsets', 'data'):
                     check_equal_but_not_same_array_attribute(attribute)
 
+    @sup_complex
+    def test_astype_immutable(self):
+        D = array([[2.0 + 3j, 0, 0],
+                   [0, 4.0 + 5j, 0],
+                   [0, 0, 0]])
+        S = self.spmatrix(D)
+        if hasattr(S, 'data'):
+            S.data.flags.writeable = False
```

Comment:
Here we only define the `data` to be read-only. However, from the original issue, we could also have read-only `intptr` and `indices`.